### PR TITLE
Integration and n+1 problem

### PR DESCRIPTION
**Pull request by kene-creator**
- N+1 problem is solved
-  Used Capybara to write integration test

### CODE REVIEW

File: spec/view/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    it 'display the how many comments' do
+      visit('/users/30/posts/')
+      expect(page).to have_content(@first_post.comments_counter)
+    end
+
+    it 'display how many Likes' do
+      visit('/users/30/posts/')
+      expect(page).to have_content('Likes: 0')
+    end
```
Comment (cynthiainga) [inline spec/view/post_index_spec.rb:42]: - [ ] Good job so far, but you should also test the next feature:
  - Tests that assert you can see the user's profile picture.
  - Tests that assert you can see the user's username.
  - Tests that assert you can see the number of posts the user has written.
  - Tests that assert you can see a section for pagination if there are more posts than fit on the view.
  - Tests that redirect to the post's show page when you click on a post.

 - [ ] Please kindly add that as well 🙃 🙃

File: spec/view/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  it 'displays theuser\'s bio ' do
+    Capybara.visit "/users/#{@user1.id}"
+    expect(page).to have_content(@user1.bio)
+  end
+
+  it 'displays 2 recent posts' do
+    Capybara.visit "/users/#{@user1.id}"
+    expect(page).to have_content(@post1.text)
+    expect(page).to have_content(@post3.text)
+  end
```
Comment (cynthiainga) [inline spec/view/user_show_spec.rb]: - [ ] Good job so far, but this is supposed to test the user's first 3 posts, please kindly fix that 🙃 
- [ ] Also you are supposed to test:
  -  that asserts you can see a button that lets me view all of a user's posts.
  - that redirects to the post's show page when you click on a user's post.

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @kene-creator, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] Descriptive PR ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- Some of your test in the requesters folder doesn’t pass, please it should be better if you fix your test and make sure all your tests are passing. I make this as optional because is not a part of the requirement in this part of the project but it good to make sure all your tests are passing 🙃 😉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @kene-creator, @karayamanemre and @NatanimA,

✔️ Tests are well written and passing

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**